coderd/batchstats/batcher.go

```diff
@@ -0,0 +1,255 @@
+package batchstats
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/sloghuman"
+
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/database/dbauthz"
+	"github.com/coder/coder/codersdk/agentsdk"
+)
+
+const (
+	// DefaultBatchSize is the default size of the batcher's buffer.
+	DefaultBatchSize = 1024
```

Comment:
In a typical environment, how many items are in the buffer?

Reply:
Assuming 1000 workspaces, the worst-case scenario is that they all attempt to push stats at once. That's extremely unlikely however.